File: config/DIConfig.java
package config;

import data.AttendanceDAO;
import service.AttendanceService;

/**
 * Dependency Injection Configuration
 */
public class DIConfig {

    public static void main(String[] args) {

        // Create DAO
        AttendanceDAO attendanceDAO = new AttendanceDAO();

        // Inject DAO into Service
        AttendanceService attendanceService =
                new AttendanceService(attendanceDAO);

        // Simulate application flow
        attendanceService.markAttendance(
                "STU101",
                "2026-01-24",
                "Present"
        );

        attendanceService.viewAttendance("STU101");
    }
}

File: config/DIConfig.py
# config/DIConfig.py

from data.AttendanceDAO import AttendanceDAO
from service.AttendanceService import AttendanceService
from presentation.controllers.AttendanceController import AttendanceController

def main():
    # Create DAO
    dao = AttendanceDAO()

    # Inject DAO → Service
    service = AttendanceService(dao)

    # Inject Service → Controller
    controller = AttendanceController(service)

    # Simulate system flow
    controller.mark_attendance("STU101", "2026-01-24", "Present")
    controller.view_attendance("STU101")

if __name__ == "__main__":
    main()
